src/services/blocks/BlocksService.ts

```diff
@@ -251,21 +265,35 @@ export class BlocksService extends AbstractService {
 							weightInfoClass
 						] as WeightPerClass).baseExtrinsic;
 				} else {
-					// We are querying a runtime that matches the decorated metadata in the api
+					// We are querying a runtime that matches the metadata in the api
 					extrinsicBaseWeight =
 						(api.consts.system?.extrinsicBaseWeight as AbstractInt) ||
 						(api.consts.system.blockWeights.perClass[
 							weightInfoClass
 						] as WeightPerClass).baseExtrinsic;
 				}
 
+				if (extrinsicBaseWeight === undefined) {
+					extrinsics[idx].info = {
+						error: 'Error occured while finding the extrinsicBaseWeight',
+					};
+					continue;
+				}
+
+				let calcFeeExBaseWeight;
+				if (typeof extrinsicBaseWeight !== 'number') {
+					calcFeeExBaseWeight = extrinsicBaseWeight.toBigInt();
+				} else {
+					calcFeeExBaseWeight = BigInt(extrinsicBaseWeight);
+				}
```

Comment:
Could this conversion be just `let calcFeeExBaseWeight = extrinsicBaseWeight.toBigInt();`? It seems we're casting to `BigInt` regardless of the value being string or number so maybe we can save a few lines of code?

Also: this happens inside a for-loop; does `extrinsicBaseWeight` really change for each extrinsic? Maybe there's some optimization here too if, say, 90% of extrinsics are balance transfers?

Reply:
I think extrinsicBaseWeight should just be persisted as a BigInt to avoid repetitive conversion.